vocabularies/Common.json

```diff
@@ -1288,6 +1288,132 @@
+            "Description": "Relative URL for invoking a function import of the [retrieval function](#Template_RecommendationsRetrievalFunction)",
+            "Value": "RetrieveRecommendations?ID=af43c2"
+          }
+        }
+      }
+    ],
+    "Template_RecommendationsRetrievalFunction": [
+      {
+        "$Kind": "Function",
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for functions that retrieve AI-based recommendations for the bound entity and its related entities",
+        "@Core.LongDescription": "The template function itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "ID",
+            "@Core.Description": "ID that occurs in the URL returned by a prior invocation of the [recommendations function](#Template_RecommendationsFunction)"
+          }
+        ],
+        "$ReturnType": {
+          "$Collection": true,
+          "$Type": "Common.PropertyRecommendationType",
+          "@Core.Description": "A collection of recommendations per targeted property"
+        }
+      }
+    ],
+    "Template_AsyncRecommendationsFunction": [
+      {
+        "$Kind": "Function",
+        "$IsBound": true,
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for functions that compute AI-based recommendations in asynchronous OData requests",
+        "@Core.LongDescription": "The template function itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "Entity",
+            "$Type": "Edm.EntityType",
+            "@Core.Description": "See parameter `Entity` in [`Template_RecommendationsFunction`](#Template_RecommendationsFunction)"
+          },
+          {
+            "$Name": "Targets",
+            "$Collection": true,
+            "@Core.IsURL": true,
+            "@Core.Description": "See parameter `Targets` in [`Template_RecommendationsFunction`](#Template_RecommendationsFunction)"
+          }
+        ],
+        "$ReturnType": {
+          "$Collection": true,
+          "$Type": "Common.PropertyRecommendationType",
+          "@Core.Description": "See return type in [`Template_RecommendationsRetrievalFunction`](#Template_RecommendationsRetrievalFunction)"
+        }
+      }
+    ],
+    "PropertyRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "target": {
+        "@Core.IsURL": true,
+        "@Core.Description": "URL (relative to the URL of the recommendations function request) addressing a property [OData-URL, section 4.6] that is targeted by this recommendation",
+        "@Core.LongDescription": "The request `SalesOrder('A')/ns.RecommendationsFunction?Targets=...`\n            might retrieve a recommendation with `\"target\": \"Items(10)/Product\"`."
+      },
+      "value": { "@Core.Description": "Recommended value, converted to string" },
```

Comment:
Why would we convert the recommended value to string ?

Reply:
Because `Collection(Edm.PrimitiveType)` is not allowed in OData.

Reply:
Isn't that a good use case for Edm.Untyped then ?

Reply:
Sorry, this is blast from the past, when `alternatives` as still a collection of primitives. But now it is a collection of a complex type, therefore we can use `Edm.PrimitiveType`. Thanks for making me aware of this!